src/rpc/masternode.cpp

```diff
@@ -587,7 +588,16 @@ static UniValue masternodelist(const JSONRPCRequest& request)
         return (int)pindex->nTime;
     };
 
+    bool showAllBannedMns = strMode != "recent";
+    int tipHeight = WITH_LOCK(cs_main, return ::ChainActive().Tip()->nHeight;);
+    constexpr int monthBlockLength = 24 * 24 * 30;
     mnList.ForEachMN(false, [&](auto& dmn) {
+        if (!showAllBannedMns && mnList.IsMNPoSeBanned(dmn)) {
```

Comment:
`showAllBannedMns` is `!= recent` and then we use `!showAllBannedMns` so it's double-negative which is a bit confusing. Could maybe refactor it to `showRecentMnsOnly` or drop the variable and simply use
```suggestion
        if (strMode == "recent" && mnList.IsMNPoSeBanned(dmn)) {
```